provisioning/service/src/Authentication/ServiceAuthenticationWithSharedAccessPolicyKey.cs

```diff
@@ -52,21 +52,13 @@ public ServiceConnectionStringBuilder Populate(ServiceConnectionStringBuilder pr
 
         private void SetPolicyName(string policyName)
         {
-            if (string.IsNullOrWhiteSpace(policyName))
-            {
-                throw new ArgumentNullException(nameof(policyName));
-            }
-
+            Argument.AssertNotNullOrWhiteSpace(policyName, nameof(policyName));
```

Comment:
Same with this. A property that calls this doesn't have a parameter. The "parameter", if you will, is `value`.

In this case I'd suggest we stop validating these values, at least at this time. We can either let it go to the service who will then throw an error, or we can check the values at a point where it matters and throw an InvalidOperationException.